tidy3d/components/simulation.py

```diff
@@ -638,6 +639,97 @@ def _projection_monitors_homogeneous(cls, val, values):
 
         return val
 
+    @pydantic.validator("monitors", always=True)
+    def _projection_direction(cls, val, values):
+        """Warn if field projection observation points are behind surface projection monitors."""
+        # This validator is in simulation.py rather than monitor.py because volume monitors are
+        # eventually converted to their bounding surface projection monitors, in which case we
+        # do not want this validator to be triggered.
```

Comment:
Hmmm. Is there some pydantic magic than can allow us to choose whether to trigger this validator if it were in the monitors @tylerflex ? I don't know if we want to go this way but it may be a bit better in the long run, e.g. consider a standalone field projection solver like we've been thinking of implementing. Well, for that we could also move this validator to `validators.py` and import it both here and in other solvers.

Reply:
Not sure exactly what kind of magic we need to implement. But I'd say in general let's use less magic. Especially since if we do a pydantic v2 port we'll have to rethink it

Reply:
I guess I'm just asking if there's a way to dynamically disable some validator when initializing a model. But we could live without it.

Reply:
Besides just an if statement in the validator that returns if triggered?  We could maybe make it a post init validation?

Reply:
Not sure I understand, what I'm asking is if I can do something like `Model(**model_dict)` and add some kwarg or set a config somehow that would skip that particular validator. This is for when we need to initialize something intrenally without triggering the warnign.

Reply:
I think anything you pass to the models init method will be present in the values dict in the validator. So you could add a flag in there. But then you'd have to pop it from values at some point

Reply:
One of these things that should work in theory but might not when trying it out but worth a shot

Reply:
If there's no pydantic native way then probably not worth it.

Reply:
I guess let's just merge this as is.

Reply:
Sounds good, I can do that now